src/datatypes/material.h

```diff
@@ -42,79 +42,107 @@
  'Diamond': 2.417 - 2.541
  */
 
-struct lightRay;
-struct intersection;
-struct color;
-struct texture;
-
-enum bsdfType {
-	emission = 0,
-	lambertian,
-	glass,
-	metal,
-	translucent,
-	transparent
-};
+#include <math.h>
 
-struct bsdf {
-	enum bsdfType type;
-	float weights;
-	bool (*bsdf)(struct intersection*, struct lightRay*, struct color*, struct lightRay*, pcg32_random_t*);
-};
+typedef struct
```

Comment:
Please use the existing vector datatypes and functions in `src/datatypes/vector.h`